### ISSUE TITLE
Add App Tracking Transparency

### ISSUE THREAD

**Issue by sche** [safe-global/safe-ios#1140]
https://developer.apple.com/documentation/apptrackingtransparency

1. We need to have one button on terms screen saying "Agree" and the other one saying "No, thanks" (or what it was before?) to close the "app terms" sheet
2. After agreeing, we present the system App Tracking alert and depending on user choice:
  - if "Allow" - then enable tracking and enable crashlytics
  - if "Ask App Not To Track" - then do not enable tracking or crashlytics
3. In the advanced app settings, we may need to rephrase it to "Share Usage and Crash Data" instead of "Share Usage Data"
4. For existing users updating to iOS 14.5, the app should ask the system permission with the same logic as above.
5. If user changes the permission to **deny** tracking in the phone/device settings, then the app must disable tracking. 
6. If the user changes permission to **allow** tracking for the app without first enabling it in the app settings, then the app still does not track, because the user would need to enable tracking in app settings.
7. If user enables the tracking in the app settings but the system permission is denied, then the app should show alert redirecting to the settings. The alert would say
  - Please allow tracking in Settings
  - "Cancel" button - closes alert
  - "Settings" button - opens device settings
  - If user then goes to settings, enables the tracking, and goes back to the app, then (if technically possible), the app is allowed to enable tracking. Please verify that this requirement and requirement 6 above do not conflict with each other.

**Comment by liliya-soroka** [comment]
Verified
app version 2.16.0 (429)
